core/src/main/scala/cats/Unfoldable.scala

```diff
@@ -0,0 +1,59 @@
+package cats
+
+import simulacrum.typeclass
+
+@typeclass trait Unfoldable[F[_]] { self =>
+
+  /**
+    * Build an F[A] from a `seed` and a generating function `f``
+    * The generating function `f` behave as follows:
+    * - If `f(b)` is `None`, then `unfoldLeft(b)(f)` should be empty.
+    * - If `f(b)` is `Some((b1, a))`, then `unfoldLeft(b)(f)` should consist of `a` appended by `unfoldLeft(b1)(f)`
+    */
+  def unfoldLeft[A, B](seed: B)(f: B => Option[(B, A)]): F[A]
```

Comment:
seems like `F[A]` may be a functor. Certainly if I have `B, B => Option[(B, A)]` to get `F[A]` and if I have `A => C` then I can make `F[C]`.

So, if for all `F[A]` there exists `B, B => Option[(B, A)]` to unfold to make `F[A]`, then `F[A]` is a functor.